test/controllers/api/v2/compute_resources_controller_test.rb

```diff
@@ -352,6 +351,25 @@ def teardown
     assert host_bm.compute?
   end
 
+  test "should associate hosts that match to a specific vm" do
+    host_cr = FactoryBot.create(:host, :on_compute_resource)
+    host_bm = FactoryBot.create(:host)
+
+    uuid = Foreman.uuid
+    vm2 = mock('vm2')
+    vm2.expects(:identity).at_least_once.returns(uuid)
+    Foreman::Model::EC2.any_instance.expects(:associated_host).returns(host_bm)
+    Foreman::Model::EC2.any_instance.expects(:find_vm_by_uuid).returns(vm2)
+    put :associate, params: { :id => host_cr.compute_resource.to_param, :vm_id => uuid }
```

Comment:
```suggestion
    put :associate, params: { :id => host_cr.compute_resource.id, :vm_id => uuid }
```
Shouldn't it be just the id?